static/app/views/onboarding/targetedOnboarding/types.ts

```diff
@@ -34,13 +34,16 @@ export type ClientState = {
   selectedPlatforms: PlatformKey[];
 };
 
-export function fetchClientState(api: Client, orgSlug: string): Promise<ClientState> {
-  return api
-    .requestPromise(`/organizations/${orgSlug}/client-state/onboarding/`)
-    .then(lastState => {
-      // Set default values
-      lastState.platformToProjectIdMap = lastState.platformToProjectIdMap || {};
-      lastState.selectedPlatforms = lastState.selectedPlatforms || [];
-      return lastState;
-    });
+export function useOnboardingState(): [
+  OnboardingState | null,
+  (next: OnboardingState | null) => void
+] {
+  const [state, setState] = usePersistedStoreCategory('onboarding');
+  const onboardingState = state
+    ? {
+        platformToProjectIdMap: state.platformToProjectIdMap || {},
+        selectedPlatforms: state.selectedPlatforms || [],
+      }
+    : null;
```

Comment:
@Neo-Zhixing would typing be easier if we returned an empty object for state instead of null?

Reply:
No because we need to distinguish between "client state is not loaded yet" versus "client state is loaded but empty".